components/selection/selection-mixin.js

```diff
@@ -12,11 +12,16 @@ const keyCodes = {
 
 export class SelectionInfo {
 
-	constructor(keys, state) {
+	constructor(keys, state, allPages) {
 		if (!keys) keys = [];
 		if (!state) state = SelectionInfo.states.none;
 		this._keys = keys;
 		this._state = state;
+		this._allPages = allPages;
+	}
+
+	get allPages() {
```

Comment:
I initially had another state for deselect all pages (`noPages`) but went with a new property to signify if the `none` or `all` state applies to all pages. It wouldn't make sense for this to be true when in the `some` state, but I don't think that can happen.

I believe the activity-evaluation-editor (the component I'm modifying this for) is the only consumer of the `select-all-pages-allowed` attribute so it's okay that I remove the `allPages` state. But let me know if you find a reference to it somewhere else.